### DISCUSSION TITLE
Create Table API

### DISCUSSION THREAD

**Discussion by MGW-SumathyV** [nocodb/nocodb#2599] [Q&A] [answered]
Hi,

I created table 'TestTable3' using below API

"http://localhost:8080/api/v1/db/meta/projects/p_llt1ueun02601z/tables"

I am getting below message in server console.

        "CREATE TABLE `TestTable3` ( `undefined` varchar(255) NULL); - Table 'testtable3' already exists"

When I open the project (nocodb dashboard) `TestTable3` is not getting displayed.

Could you please solve this

**Comment by wingkwong** [comment]
Can you provide the request body as well?

**Reply by MGW-SumathyV** [reply]
data: {                     
                          "id": "string",
                          "fk_project_id": "string",
                          "fk_base_id": "string",
                          "table_name": "TestTable3",
                          "title": "TestTable3",
                          "type": "string",
                          "enabled": "string",
                          "parent_id": "string",
                          "show_as": "string",
                          "tags": "string",
                          "pinned": true,
                          "deleted": true,
                          "order": 0,
                          "columns": [
                            {
                              "uidt": "1"
                            }
                          ]
                        }

**Comment by wingkwong** [comment, marked as answer]
Can you also check the following - 

1. Go to your DB to see if `TestTable3` exists
1.1. If so, go to [Sync Page](https://docs.nocodb.com/setup-and-usages/sync-schema) to see `Sync` is available.
2. Try to create a table using API with the table name which doesn't exist in your DB

**Reply by MGW-SumathyV** [reply]
Thank you. Its reflecting now.
I tried creating a table with new Table name (request body as below). But getting "Cannot read property 'uidt' of undefined" 
in server console .However table is getting created.

 data: {
                       "id": "string",
                          "fk_project_id": "string",
                          "fk_base_id": "string",
                          "table_name": "TestTable4",
                          "title": "TestTable4",
                          "type": "string",
                          "enabled": "string",
                          "parent_id": "string",
                          "show_as": "string",
                          "tags": "string",
                          "pinned": true,
                          "deleted": true,
                          "order": 0,
                          "columns": [
                            {
                              "uidt": "1"                       
                            }
                          ]
                        }

Can "uidt" value be any string? Is it a unique identifier?

**Reply by wingkwong** [reply]
`uidt` means UI data type. Some examples would be `SingleLineText`, `DateTime`, `ID` and ertc. But it's more that that to fill in `columns`. I suggest you can open network tab to see the payload when you create a table in dashboard.

**Reply by MGW-SumathyV** [reply]
Thank you for your support.
